dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtil.java

```diff
@@ -106,24 +111,42 @@ public boolean isDataBaseContainerId(final String containerIdentifier) {
     }
 
     public Host getHost (final String path) throws DotSecurityException, DotDataException {
-
         return this.getHostFromHostname(this.getHostName(path));
     }
 
     public Host getHostFromHostname (final String hostname) throws DotSecurityException, DotDataException {
-
         return null == hostname?
-                APILocator.getHostAPI().resolveHostName(hostname, APILocator.systemUser(), false):
-                APILocator.getHostAPI().findDefaultHost(APILocator.systemUser(), false);
+                APILocator.getHostAPI().findDefaultHost(APILocator.systemUser(), false) :
+                APILocator.getHostAPI().resolveHostName(hostname, APILocator.systemUser(), false);
     }
 
     //demo.dotcms.com/application/containers/test/
-    public String getHostName (final String path) {
+    public String getHostName(final String path) {
+        try {
+            String tmp = path;
+            final List<String> pageModePrefixList = Stream.of(PageMode.values())
+                    .map(pageMode -> String.format("/%s/", pageMode.name()))
+                    .collect(Collectors.toList());
+            for (final String prefix : pageModePrefixList) {
+                if (tmp.startsWith(prefix)) {
+                    tmp = tmp.substring(prefix.length());
+                    break;
+                }
+            }
 
-        final int startsHost = null != path?     path.indexOf(HOST_INDICATOR): -1;
-        final int endsHost   = -1 != startsHost? path.indexOf(FORWARD_SLASH, startsHost+HOST_INDICATOR.length()): -1;
-        return startsHost != -1 && endsHost != -1?
-                path.substring(startsHost+HOST_INDICATOR.length(), endsHost): null;
+            tmp = tmp.replaceAll(HOST_INDICATOR, "");
+            tmp = tmp.substring(0, tmp.indexOf(CONTAINER_FOLDER_PATH));
+            final String finalString = tmp;
+            Logger.warn(FileAssetContainerUtil.class,
+                    () -> String.format(" extracted hostName `%s`", finalString));
+
+            return (UtilMethods.isSet(tmp) ? tmp : null);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4599597)